tests/ui/unreachable-code/type_ascribe_never_field.rs

```diff
@@ -0,0 +1,17 @@
+// Regression test for <https://github.com/rust-lang/rust/issues/149431>
```

Comment:
In the previous commit, you moved tests from `unreachable-code` to `reachable`, but this test is in `unreachable-code`?